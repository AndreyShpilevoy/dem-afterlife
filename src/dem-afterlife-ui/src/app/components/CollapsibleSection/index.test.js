/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import {CollapsibleSectionPure} from './index';

describe('CollapsibleSection Pure', () => {
    const classes = {
        '.default-CollapsibleSection-bodyHolder.closed': 'default-CollapsibleSection-bodyHolder.closed',
        '.default-CollapsibleSection-headerArrow > .SVGInline-svg': 'default-CollapsibleSection-headerArrow > .SVGInline-svg',
        '.default-CollapsibleSection-headerArrow.closed': 'default-CollapsibleSection-headerArrow.closed',
        body: 'default-CollapsibleSection-body',
        bodyHolder: 'default-CollapsibleSection-bodyHolder',
        from: 'from',
        general: 'default-CollapsibleSection-general',
        header: 'default-CollapsibleSection-header',
        headerArrow: 'default-CollapsibleSection-headerArrow',
        headerArrowHolder: 'default-CollapsibleSection-headerArrowHolder',
        headerColumn: 'default-CollapsibleSection-headerColumn',
        headerCursor: 'default-CollapsibleSection-headerCursor',
        headerHolder: 'default-CollapsibleSection-headerHolder',
        headerText: 'default-CollapsibleSection-headerText',
        termedSection: 'default-CollapsibleSection-termedSection',
        titleClass: 'default-CollapsibleSection-titleClass',
        to: 'to'
    };

    it('without headerSettings', () => {
        expect(shallow(<CollapsibleSectionPure classes={classes}><div>{'Column content'}</div></CollapsibleSectionPure>)).toMatchSnapshot();
    });

    it('without headerSettings', () => {
        const wrapper = shallow(<CollapsibleSectionPure classes={classes}><div>{'Column content'}</div></CollapsibleSectionPure>);
        wrapper.find('Aesthetic-RowPure').first().simulate('click');
        expect(wrapper).toMatchSnapshot();
    });

    it('with headerSettings', () => {
        const headerSettings = {
            title: 'title',
            firstColumnTerm: 'firstColumnTerm',
            secondColumnTerm: 'secondColumnTerm',
            thirdColumnTerm: 'thirdColumnTerm'
        };
        expect(shallow(
            <CollapsibleSectionPure classes={classes} headerSettings={headerSettings}>
                <div>
                    {'Column content'}
                </div>
            </CollapsibleSectionPure>
        )).toMatchSnapshot();
    });

    it('not default collapseSettings', () => {
        const collapseSettings = {
            collapsedByDefault: true,
            isCollapsible: false
        };
        expect(shallow(
            <CollapsibleSectionPure classes={classes} collapseSettings={collapseSettings}>
                <div>
                    {'Column content'}
                </div>
            </CollapsibleSectionPure>
        )).toMatchSnapshot();
    });
});
