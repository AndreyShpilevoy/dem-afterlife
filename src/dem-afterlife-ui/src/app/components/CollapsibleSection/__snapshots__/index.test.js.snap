// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleSection Pure after click on header state should be changed 1`] = `
<div
  className="default-CollapsibleSection-general"
>
  <Jss(RowPure)
    className="default-CollapsibleSection-header"
    onClick={[Function]}
    reverse={false}
  >
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerText default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={11}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <span
        className="default-CollapsibleSection-titleClass"
      />
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={1}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <Jss(HiddenPure)
        className="default-CollapsibleSection-termedSection"
        lg=""
        md="down"
        sm=""
        xl=""
        xs=""
      >
        <Jss(RowPure)
          className=""
          reverse={false}
        >
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={6}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
        </Jss(RowPure)>
      </Jss(HiddenPure)>
      <Jss(HiddenPure)
        className="default-CollapsibleSection-headerArrowHolder"
        lg="up"
        md=""
        sm="down"
        xl=""
        xs=""
      >
        <SvgIconsMapper
          className="default-CollapsibleSection-headerArrow"
          iconName="SortLeft"
        />
      </Jss(HiddenPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-CollapsibleSection-bodyHolder default-CollapsibleSection-headerCursor"
  >
    <div
      className="default-CollapsibleSection-body"
    >
      <div>
        Column content
      </div>
    </div>
  </div>
</div>
`;

exports[`CollapsibleSection Pure not default collapseSettings 1`] = `
<div
  className="default-CollapsibleSection-general"
>
  <Jss(RowPure)
    className="default-CollapsibleSection-header"
    onClick={[Function]}
    reverse={false}
  >
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerText default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={11}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <span
        className="default-CollapsibleSection-titleClass"
      />
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={1}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <Jss(HiddenPure)
        className="default-CollapsibleSection-termedSection"
        lg=""
        md="down"
        sm=""
        xl=""
        xs=""
      >
        <Jss(RowPure)
          className=""
          reverse={false}
        >
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={6}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
        </Jss(RowPure)>
      </Jss(HiddenPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-CollapsibleSection-bodyHolder closed"
  >
    <div
      className="default-CollapsibleSection-body"
    >
      <div>
        Column content
      </div>
    </div>
  </div>
</div>
`;

exports[`CollapsibleSection Pure with headerSettings 1`] = `
<div
  className="default-CollapsibleSection-general"
>
  <Jss(RowPure)
    className="default-CollapsibleSection-header"
    onClick={[Function]}
    reverse={false}
  >
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerText default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={11}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <span
        className="default-CollapsibleSection-titleClass"
      >
        title
      </span>
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={1}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <Jss(HiddenPure)
        className="default-CollapsibleSection-termedSection"
        lg=""
        md="down"
        sm=""
        xl=""
        xs=""
      >
        <Jss(RowPure)
          className=""
          reverse={false}
        >
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            >
              firstColumnTerm
            </div>
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            >
              secondColumnTerm
            </div>
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={6}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            >
              thirdColumnTerm
            </div>
          </Jss(ColumnPure)>
        </Jss(RowPure)>
      </Jss(HiddenPure)>
      <Jss(HiddenPure)
        className="default-CollapsibleSection-headerArrowHolder"
        lg="up"
        md=""
        sm="down"
        xl=""
        xs=""
      >
        <SvgIconsMapper
          className="default-CollapsibleSection-headerArrow"
          iconName="SortLeft"
        />
      </Jss(HiddenPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-CollapsibleSection-bodyHolder default-CollapsibleSection-headerCursor"
  >
    <div
      className="default-CollapsibleSection-body"
    >
      <div>
        Column content
      </div>
    </div>
  </div>
</div>
`;

exports[`CollapsibleSection Pure without headerSettings 1`] = `
<div
  className="default-CollapsibleSection-general"
>
  <Jss(RowPure)
    className="default-CollapsibleSection-header"
    onClick={[Function]}
    reverse={false}
  >
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerText default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={11}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <span
        className="default-CollapsibleSection-titleClass"
      />
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={1}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <Jss(HiddenPure)
        className="default-CollapsibleSection-termedSection"
        lg=""
        md="down"
        sm=""
        xl=""
        xs=""
      >
        <Jss(RowPure)
          className=""
          reverse={false}
        >
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={6}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
        </Jss(RowPure)>
      </Jss(HiddenPure)>
      <Jss(HiddenPure)
        className="default-CollapsibleSection-headerArrowHolder"
        lg="up"
        md=""
        sm="down"
        xl=""
        xs=""
      >
        <SvgIconsMapper
          className="default-CollapsibleSection-headerArrow"
          iconName="SortLeft"
        />
      </Jss(HiddenPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-CollapsibleSection-bodyHolder default-CollapsibleSection-headerCursor"
  >
    <div
      className="default-CollapsibleSection-body"
    >
      <div>
        Column content
      </div>
    </div>
  </div>
</div>
`;

exports[`CollapsibleSection Pure without headerSettings 2`] = `
<div
  className="default-CollapsibleSection-general"
>
  <Jss(RowPure)
    className="default-CollapsibleSection-header"
    onClick={[Function]}
    reverse={false}
  >
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerText default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={11}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <span
        className="default-CollapsibleSection-titleClass"
      />
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-CollapsibleSection-headerHolder"
      lg={6}
      lgOffset=""
      md={1}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs=""
      xsOffset=""
    >
      <Jss(HiddenPure)
        className="default-CollapsibleSection-termedSection"
        lg=""
        md="down"
        sm=""
        xl=""
        xs=""
      >
        <Jss(RowPure)
          className=""
          reverse={false}
        >
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={3}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
          <Jss(ColumnPure)
            className="default-CollapsibleSection-headerColumn"
            lg={6}
            lgOffset=""
            md=""
            mdOffset=""
            sm=""
            smOffset=""
            xl=""
            xlOffset=""
            xs=""
            xsOffset=""
          >
            <div
              className="default-CollapsibleSection-headerText"
            />
          </Jss(ColumnPure)>
        </Jss(RowPure)>
      </Jss(HiddenPure)>
      <Jss(HiddenPure)
        className="default-CollapsibleSection-headerArrowHolder"
        lg="up"
        md=""
        sm="down"
        xl=""
        xs=""
      >
        <SvgIconsMapper
          className="default-CollapsibleSection-headerArrow"
          iconName="SortLeft"
        />
      </Jss(HiddenPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-CollapsibleSection-bodyHolder default-CollapsibleSection-headerCursor"
  >
    <div
      className="default-CollapsibleSection-body"
    >
      <div>
        Column content
      </div>
    </div>
  </div>
</div>
`;
